consensus/state.go

```diff
@@ -756,17 +756,32 @@ func (cs *ConsensusState) handleTimeout(ti timeoutInfo, rs cstypes.RoundState) {
 func (cs *ConsensusState) handleTxsAvailable() {
 	cs.mtx.Lock()
 	defer cs.mtx.Unlock()
-	// we only need to do this for round 0
-	cs.enterNewRound(cs.Height, 0)
-	cs.enterPropose(cs.Height, 0)
+
+	if cs.config.SkipTimeoutCommit && cs.LastCommit != nil && cs.LastCommit.HasAll() {
+		// we only need to do this for round 0
+		cs.enterNewRound(cs.Height, 0)
+		cs.enterPropose(cs.Height, 0)
+		return
+	}
+
+	now := tmtime.Now()
+	if cs.StartTime.After(now) {
```

Comment:
Can we more concretely assert that we're in the timeout commit phase here using the state/step rather than the time?